frontend/src/components/accountPage/UserListLogic/UserListLogic.tsx

```diff
@@ -0,0 +1,192 @@
+import { FC, useState, useEffect, useRef } from 'react';
+import { Spin } from 'antd';
+import {
+  useTenantsQuery,
+  useApplyTenantMutation,
+} from '../../../generated-types';
+import { getTenantPatchJson } from '../../../graphql-components/utils';
+import UserList from '../UserList/UserList';
+import { User } from '../../../utils';
+
+export interface IUserListLogicProps {
+  workspaceName: string;
+}
+
+const UserListLogic: FC<IUserListLogicProps> = props => {
+  const { workspaceName } = props;
+  const [abortUploading, setAbortUploading] = useState<boolean>(false);
+  const abortUploadingRef = useRef(false);
+  const [uploadedNumber, setUploadedNumber] = useState<number>(0);
+  const [users, setUsers] = useState<User[]>([]);
+  const { data, loading, error } = useTenantsQuery({
+    variables: {
+      labels: `crownlabs.polito.it/${workspaceName}`,
+      retrieveWorkspaces: true,
+    },
+
+    notifyOnNetworkStatusChange: true,
+  });
+
+  useEffect(() => {
+    abortUploadingRef.current = abortUploading;
+  }, [abortUploading]);
+
+  const handleAbort = (value: boolean) => setAbortUploading(value);
+  useEffect(() => {
+    if (!loading) {
+      setUsers(
+        data?.tenants?.items?.map(user => {
+          return {
```

Comment:
Use the compact form:
`stuff(something => ({ key: val }))` instead of `stuff(something => { return { key: val } })`